### PR TITLE
[Spring Beans] LPS-201042 

### PR DESCRIPTION
**Pull request by JanisZhang**
Move RawMetadataProcessorMessageListener to the module and register Destination in the activate method.
https://liferay.atlassian.net/browse/LPS-201042

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:sf

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:help

**Comment by dantewang** [comment]
ci:test:document

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/messaging/RawMetadataProcessorMessageListener.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+@Component(
+	property = "destination.name=" + DestinationNames.DOCUMENT_LIBRARY_RAW_METADATA_PROCESSOR,
+	service = MessageListener.class
+)
+public class RawMetadataProcessorMessageListener extends BaseMessageListener {
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		DestinationConfiguration destinationConfiguration =
```
Comment (dantewang) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/messaging/RawMetadataProcessorMessageListener.java:40]: @JanisZhang Also inline this. By the way, there are multiple PRs modifying messaging-misc-spring.xml, they will conflict with each other if one is merged, right?

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by JanisZhang** [comment]
@dantewang Force-pushed. I only inline the variable. Thanks!

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@JanisZhang No regressions found.

**Comment by dantewang** [comment]
@shuyangzhou Does this pattern (making the message listener a component and register destination in `activate` method) look good to you? If so we'll apply to other beans in `messaging-misc-spring.xml`.

There's another pattern in portal which uses self-activated components to register both message listener and destination but that should be what we want to avoid.

**Comment by shuyangzhou** [comment]
Yes, this is fine, please go ahead with it.

**Comment by JanisZhang** [comment]
Hi @dantewang, I've restructured the commits to show the changes as 'moved from...' rather than 'created file and deleted,' and I've force-pushed these alterations. I'll apply this to my other pending PR later. Thanks!

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/144105

@JanisZhang Please trigger tests once you've updated the other PRs of this pattern.

**Pull request closed without merging**